common/credCache_bsd.go

```diff
@@ -0,0 +1,121 @@
+// +build freebsd
+
+// Copyright © 2017 Microsoft <wastore@microsoft.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package common
+
+import (
+	"fmt"
+	"sync"
+)
+
+// CredCache manages credential caches.
+// This is just a dumb same-process memoizer for now.
+type CredCache struct {
+	keyName string // the Name of key would be cached in keyring, composed with current UID, in case user su
```

Comment:
The background here is that `credCache_bsd.go` started as a verbatim copy of `credCache_linux.go`, with pieces removed to get to the PR version.  The same comment is present on the corresponding `CredCache::keyName` member in the Linux version, and I think the comment is approximately just as valid here (which is to say, I have no idea, but the Linux credCache doesn't appear to do anything with uids either).

The `keyName` is provided in `NewCredCache()` (generated externally to the credCache implementation).  The direct external user is common/oauthTokenManager.go, `NewUserOAuthTokenManagerInstance()`.  But that's just a pass-through from cmd/credentialUtil.go, `GetUserOAuthTokenManagerInstance()`, where it appears `KeyName` is always `oauthLoginSessionCacheKeyName` or just `"AzCopyOAuthTokenCache"`.

So maybe the comment is stale in the Linux version as well?  Or perhaps the uid bits all happen in the `keyctl` package, and that's what the comment refers to.

I'm happy to remove the comment, since the parts about caching in keyring definitely don't apply to this implementation, and I don't know what the rest of it means.  I'll plan to do that.